### ISSUE TITLE
Implementing searching for keys on keyserver

### ISSUE THREAD

**Issue by bnvk** [mailvelope/mailvelope#319]
There is a pretty nifty library built by @diafygi called [PublicKey.js](https://github.com/diafygi/publickeyjs) that allows for searching the keyservers in the browser. Seems like it could be implemented in Mailvelope!

**Comment by diafygi** [comment]
SKS 1.1.5 [added CORS support](https://lists.gnupg.org/pipermail/gnupg-users/2014-May/049682.html), so you can now make requests to the current version SKS keyserver subpool from the browser. Here's the request code if you just want to request from the SKS pool (and not Keybase).

```javascript
var xhr = new XMLHttpRequest();
xhr.open("GET", "http://subset.pool.sks-keyservers.net:11371/pks/lookup?op=index&options=mr&search=diafygi");
xhr.onload = function(){
    console.log(this.responseText);
}
xhr.send();
```

Unfortunately, the SKS DNS pool can't have https (since it's a DNS round robin), so you have to choose a specific https-enabled keyserver if you want to use PublicKey.js on an https page (I use `https://keys.fedoraproject.org/` by default).

**Comment by rubo77** [comment]
As long as this is not implemented you should at least add a link to a public keysearch page if there is no key found for the receipient.

Like http://www.heise.de/security/dienste/Keyserver-474468.html

**Comment by Manouchehri** [comment]
Duplicate of #10 and #149.

**Comment by rugk** [comment]
What about hkps://hkps.pool.sks-keyservers.net? It has HTTPS support and uses the public key.

**Comment by Manouchehri** [comment]
OpenKeychain uses both HKPS and Keybase, so their code might be a good resource if you don't mind reading Java.

https://github.com/open-keychain/open-keychain/tree/master/OpenKeychain/src/main/java/org/sufficientlysecure/keychain/keyimport

**Comment by diafygi** [comment]
@rugk you'd have to add the hkps certificate authority to the root store (not sure if that's even possible in a plugin) to trust the certificates returned by keyservers on that port.

**Comment by rugk** [comment]
> @rugk you'd have to add the hkps certificate authority to the root store (not sure if that's even possible in a plugin) to trust the certificates returned by keyservers on that port.

Indeed that's a problem - unless you can maybe just pin the cert (hash).
So possibly using HTTPS and the usual web interface or something similar is a better idea.

**Comment by tanx** [comment]
I started implementing a HKP js client today. We will be using this in Whiteout and in Mailvelope. 

@diafygi I tested CORS support in SKS 1.1.5 and unfortunately it does not work. See:
https://lists.gnupg.org/pipermail/gnupg-users/2015-December/054821.html

**Comment by diafygi** [comment]
@tanx I just tried your link and it worked on both Chrome and Firefox via ajax. Can you please give more details about what you've tested? Here's my test code (copy and paste it into your console on https://keys.whiteout.io).

```
var xhr = new XMLHttpRequest();
xhr.open("GET", "https://keyserver.ubuntu.com/pks/lookup?op=get&options=mr&search=safewithme.testuser%40gmail.com");
xhr.onload = function(){ console.log(xhr.responseText); };
xhr.send();
```

**Comment by tanx** [comment]
@diafygi you are right. Sorry my bad. It seems like I just had problems with my connection yesterday. I've added initial HKP support to OpenPGP.js. Feel free to comment: https://github.com/openpgpjs/openpgpjs/pull/380

Thanks!

**Comment by gfairchild** [comment]
I'd love to see this implemented.

**Comment by DreamFlasher** [comment]
Isn't this implemented?
